src/main/java/frc/robot/subsystems/Shooter.java

```diff
@@ -5,51 +5,187 @@
+import com.ctre.phoenix6.controls.DutyCycleOut;
+import com.ctre.phoenix6.controls.Follower;
+import com.ctre.phoenix6.controls.VelocityTorqueCurrentFOC;
+import com.ctre.phoenix6.controls.VelocityVoltage;
 import com.ctre.phoenix6.hardware.TalonFX;
+import edu.wpi.first.math.controller.SimpleMotorFeedforward;
 import edu.wpi.first.wpilibj.smartdashboard.SmartDashboard;
 import edu.wpi.first.wpilibj2.command.SubsystemBase;
 import frc.robot.constants.CAN;
+import frc.robot.constants.FLYWHEEL;
+import frc.robot.utils.CtreUtils;
+import org.littletonrobotics.junction.Logger;
 
 public class Shooter extends SubsystemBase {
-  private final TalonFX flywheelmotor1 = new TalonFX(CAN.flywheel1);
-  private final TalonFX flywheelmotor2 = new TalonFX(CAN.flywheel2);
+
   private double m_rpm;
+  private boolean m_testMode = false;
+  private double m_headingOffset;
   private double flywheelRPMRatio = 1.0;
+  private double m_percentOutput;
+
+  private final TalonFX[] m_shooterMotors = {
+    new TalonFX(CAN.flywheel1), new TalonFX(CAN.flywheel2) // Flywheel[0] is bottom
+  };
+
+  private double flywheelPercentRatio = 1.0;
+  private final DutyCycleOut m_dutyCycleRequest = new DutyCycleOut(0);
+  private final VelocityVoltage m_velocityRequest = new VelocityVoltage(0);
+
+  private final SimpleMotorFeedforward m_feedForward =
+      new SimpleMotorFeedforward(FLYWHEEL.kS, FLYWHEEL.kV, FLYWHEEL.kA);
+  private SimpleMotorFeedforward m_currentFeedForward = m_feedForward;
+
+  VelocityTorqueCurrentFOC FOCcontrolBottom = new VelocityTorqueCurrentFOC(0);
+
+  VelocityTorqueCurrentFOC FOCcontrolTop = new VelocityTorqueCurrentFOC(0);
 
   // private final ConfigFactoryDefault configSelectedFeedbackSensor = new Config
   /* Creates a new Intake. */
   public Shooter() {
+    TalonFXConfiguration configBottom = new TalonFXConfiguration();
+
+    configBottom.Feedback.SensorToMechanismRatio = FLYWHEEL.gearRatioBottom;
+    CtreUtils.configureTalonFx(m_shooterMotors[0], configBottom);
+
+    TalonFXConfiguration configTop = new TalonFXConfiguration();
+    configBottom.Feedback.SensorToMechanismRatio = FLYWHEEL.gearRatioTop;
+    CtreUtils.configureTalonFx(m_shooterMotors[1], configTop);
+
+    configBottom.Slot0.kP = FLYWHEEL.kP;
+    configBottom.Slot0.kI = FLYWHEEL.kI;
+    configBottom.Slot0.kD = FLYWHEEL.kD;
+
+    configTop.Slot0.kV = FLYWHEEL.kV;
+    configTop.Slot0.kP = FLYWHEEL.kP;
+    configTop.Slot0.kI = FLYWHEEL.kI;
+    configTop.Slot0.kD = FLYWHEEL.kD;
+
+    m_shooterMotors[0].setInverted(true);
     // flywheel motor 1
-    TalonFXConfiguration config = new TalonFXConfiguration();
-    flywheelmotor1.getConfigurator().apply(config);
-    flywheelmotor1.setInverted(true);
-    // flywheel motor 2
-    flywheelmotor2.getConfigurator().apply(config);
-    flywheelmotor2.setInverted(false);
+    m_shooterMotors[1].setControl(new Follower(m_shooterMotors[0].getDeviceID(), true));
   }
 
   // values that we set
```

Comment:
Make comment above this function: Only use when absolutely needed. For every other case use RPM.